x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/settings/InternalServiceSettings.java

```diff
@@ -81,10 +81,15 @@ public int hashCode() {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
```

Comment:
what happens to this builder object? does it get closed as the fragmentBuilder?

Reply:
if they are the same object, maybe we should just discard the return value? its a bit confusing to have two variables which reference the same object in the same context.

Reply:
Maybe the toXContentFragment function could be renamed to `addXContentFragment` or something, to better indicate whats happing? Or is that function an override?